services/explorer-indexer/src/modules/dag/dag.service.ts

```diff
@@ -41,35 +43,67 @@ export default class DagService {
       transactionCount,
       transactions,
     } = { ...dagData };
-    const cherryPick: IDAG = {
-      hash: zeroX(hash),
-      pivot: zeroX(pivot),
-      tips,
-      level: parseInt(`${level}`, 16) || 0,
-      pbftPeriod: period,
-      timestamp: parseInt(timestamp, 16) || 0,
-      author: zeroX(author),
-      signature: zeroX(signature),
-      vdf: parseInt(vdf?.difficulty, 16) || 0,
-      transactionCount,
+
+    const txes = await findTransactionsByHashesOrFill(
       transactions,
-    };
-    const saved = await this.dagRepository.save(cherryPick);
-    if (saved) {
-      this.logger.log(`Registered new DAG ${saved.hash}`);
+      this.txRepository,
+      this.logger
+    );
+    let newDag = await this.dagRepository.findOneBy({ hash: zeroX(hash) });
+    if (newDag) {
+      newDag.pivot = zeroX(pivot);
+      newDag.tips = tips;
+      newDag.level = parseInt(`${level}`, 16) || 0;
+      newDag.pbftPeriod = period;
+      newDag.timestamp = parseInt(timestamp, 16) || 0;
+      newDag.author = zeroX(author);
+      newDag.signature = zeroX(signature);
+      newDag.vdf = parseInt(vdf?.difficulty, 16) || 0;
+      newDag.transactionCount = transactionCount;
+      newDag.transactions = txes;
+    } else {
+      newDag = new DagEntity({
+        hash: zeroX(hash),
+        pivot: zeroX(pivot),
+        tips,
+        level: parseInt(`${level}`, 16) || 0,
+        pbftPeriod: period,
+        timestamp: parseInt(timestamp, 16) || 0,
+        author: zeroX(author),
+        signature: zeroX(signature),
+        vdf: parseInt(vdf?.difficulty, 16) || 0,
+        transactionCount,
+        transactions: txes,
+      });
     }
+    try {
+      const saved = await this.dagRepository.save(newDag);
+      if (saved) {
+        this.logger.log(`Registered new DAG ${saved.hash}`);
+      }
 
-    return saved;
+      return saved;
+    } catch (error) {
+      console.error('handleNewDag');
+      console.error(error);
+    }
   }
 
   public async updateDag(updateData: NewDagBlockFinalizedResponse) {
     let dag;
 
-    dag = await this.dagRepository.findOneBy({ hash: updateData.block });
-    if (!dag) dag = new DagEntity();
-    dag.hash = zeroX(updateData.block);
+    dag = await this.dagRepository.findOneBy({ hash: zeroX(updateData.block) });
+    if (!dag) {
+      dag = new DagEntity();
+      dag.hash = zeroX(updateData.block);
+    }
     dag.pbftPeriod = parseInt(updateData.period, 10);
-    const updated = await this.dagRepository.save(dag);
-    if (updated) this.logger.log(`DAG ${updateData.block} finalized`);
+    try {
+      const updated = await this.dagRepository.save(dag);
+      if (updated) this.logger.log(`DAG ${updateData.block} finalized`);
+    } catch (error) {
+      console.error('NewDagBlockFinalizedResponse');
```

Comment:
Can you add both in a single console.error?  console.error('NewDagBlockFinalizedResponse', error)